org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/codeassist/DOMCompletionEngine.java

```diff
@@ -1578,6 +1662,104 @@ public void complete(org.eclipse.jdt.internal.compiler.env.ICompilationUnit sour
 		}
 	}
 
+	private CompletionProposal toCastedReceiverProposal(IBinding castedBinding, ITypeBinding castedType, ASTNode toCast) {
+		
+		DOMInternalCompletionProposal template = (DOMInternalCompletionProposal)toProposal(castedBinding);
+		DOMInternalCompletionProposal res = null;
+		if (castedBinding instanceof IMethodBinding) {
+			res = createProposal(CompletionProposal.METHOD_REF_WITH_CASTED_RECEIVER);
+		} else if (castedBinding instanceof IVariableBinding) {
+			res = createProposal(CompletionProposal.FIELD_REF_WITH_CASTED_RECEIVER);
+		} else {
+			return template;
+		}
+		
+		// build completion
+		StringBuilder completion = new StringBuilder();
```

Comment:
There is a test case where inline comments are interspersed between the identifiers and `.`; this code breaks it. I'll likely need to use the existing CU text content into order to build the correct completion text